From 574efa2f56c2236bf495dc74240a94201e9207b9 Mon Sep 17 00:00:00 2001
From: Jordan Hand <jorhand@linux.microsoft.com>
Date: Sun, 14 Jun 2020 19:14:17 -0700
Subject: [PATCH 2/5] software_node: obtain swnode references in graph helpers

A few of the helper functions in the swnode graph API implementations
get references to the software_node_fwnode but do not obtain a reference
to the underlying kobj.

Obtain references to properly manage the lifetime of the software node.

Signed-off-by: Jordan Hand <jorhand@linux.microsoft.com>
---
 drivers/base/swnode.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/base/swnode.c b/drivers/base/swnode.c
index 7359b3f4e5da..68d3d395b294 100644
--- a/drivers/base/swnode.c
+++ b/drivers/base/swnode.c
@@ -450,7 +450,7 @@ software_node_get_next_child(const struct fwnode_handle *fwnode,
 		c = list_next_entry(c, entry);
 	else
 		c = list_first_entry(&p->children, struct swnode, entry);
-	return &c->fwnode;
+	return fwnode_handle_get(&c->fwnode);
 }
 
 static struct fwnode_handle *
@@ -465,8 +465,7 @@ software_node_get_named_child_node(const struct fwnode_handle *fwnode,
 
 	list_for_each_entry(child, &swnode->children, entry) {
 		if (!strcmp(childname, kobject_name(&child->kobj))) {
-			kobject_get(&child->kobj);
-			return &child->fwnode;
+			return fwnode_handle_get(&child->fwnode);
 		}
 	}
 	return NULL;
-- 
2.17.1

